deno/fhirchat-demo/islands/ChatIsland.tsx

```diff
@@ -25,32 +32,69 @@ export default function ChatIsland() {
     }
   }
 
+  const handleUserKeyPress = (event: KeyboardEvent) => {
+    const { key } = event;
+    if (key === "Enter" && question.value !== "") handleSubmit(event);
+    else return;
+  };
+
+  useEffect(() => {
+    globalThis.addEventListener("keydown", handleUserKeyPress);
+    return () => {
+      globalThis.removeEventListener("keydown", handleUserKeyPress);
+    };
+  });
+
   useEffect(() => {
-    console.log("use effect!");
     return () => {
       closeConversation();
     };
   }, []);
 
+  useEffect(() => {
+    if (conversation.value[1]) resetTextAnimation();
+  }, [conversation.value]);
+
   return (
-    <section class="section 
-      section-padding-large 
-      is-flex 
-      is-flex-direction-column 
-      is-justify-content-center 
-      is-align-self-center">
-      <ul>
-        {conversation.value.map((message) => <li>{message}</li>)}
-      </ul>
+    <section class="is-flex is-flex-direction-column is-justify-content-center  is-align-self-center">
+      {conversation.value[0] && (
+        <ul>
+          <li>
+            <div class="is-flex is-flex-direction-row pb-5">
+              <span class="icon is-medium">
+                <img src={"../images/user-avatar.svg"} alt="user avatar" />
+              </span>
+              <p class="is-size-6 has-text-left has-text-weight-normal pl-5 is-align-self-center user_prompt">
+                {conversation.value[0]}
+              </p>
+            </div>
+          </li>
+        </ul>
+      )}
+
+      {conversation.value[1] && (
+        <ul>
+          <li class="fhir_agent_container">
+            <div class="is-flex is-align-items-center is-flex-direction-row">
+              <span class="icon is-medium">
+                <img src={"../images/agent-avatar.svg"} alt="user avatar" />
+              </span>
+              <p class="is-size-6 has-text-left has-text-weight-normal pl-5 is-align-self-center fhir_agent_prompt">
+                {displayText}
+              </p>
+            </div>
+          </li>
+        </ul>
+      )}
```

Comment:
accessing conversation by index is temporary for demo purposes, I am looking for a better way to do this...

Reply:
@xeniyandkn we should maybe sync further on this, the conversation is intentionally an array of messages that gets appended.  Because, more than one _question & answer sequence_ can happen within the same _conversational session_.

You can continue to use the index of the first one only for the purpose of _styling_ but be aware that this won't work with the real backend and the _conversation context_ should be taken forward with that in mind.